contrib/experimental/great_expectations_experimental/expectations/expect_column_pair_values_to_have_difference_of_custom_percentage.py

```diff
@@ -0,0 +1,155 @@
+                    "out": {
+                        "success": False,
+                    },
+                },
+            ],
+        }
+    ]
+    # Setting necessary computation metric dependencies and defining kwargs, as well as assigning kwargs default values
+    success_keys = (
+        "percentage",
+        "mostly",
+    )
+
+    default_kwarg_values = {
+        "row_condition": None,
+        "condition_parser": None,
+        "percentage": 0.1,
+        "mostly": 1.0,
+        "result_format": "BASIC",
+        "include_config": True,
+        "catch_exceptions": False,
+    }
+
+
+    def validate_configuration(
+        self, configuration: Optional[ExpectationConfiguration]
+    ) -> None:
+        """
+        Validates that a configuration has been set, and sets a configuration if it has yet to be set. Ensures that
+        necessary configuration arguments have been provided for the validation of the expectation.
+        Args:
+            configuration (OPTIONAL[ExpectationConfiguration]): \
+                An optional Expectation Configuration entry that will be used to configure the expectation
+        Returns:
+            None. Raises InvalidExpectationConfigurationError if the config is not validated successfully
+        """
+        super().validate_configuration(configuration)
+        
+        mostly = configuration.kwargs["mostly"]
+        percentage = configuration.kwargs["percentage"]
+        
+        if configuration is None:
+            configuration = self.configuration
+
+        #Check if both columns are provided and values of mostly and percentage are correct
+        try:
+            assert (
+                "column_A" in configuration.kwargs
+                and "column_B" in configuration.kwargs
+            ), "both columns must be provided"
+            assert (
+                0 <= mostly <= 1
+            ), "Mostly must be between 0 and 1"
+            assert (
+                percentage >= 0
+            ), "Percentage must be positive"
+        except AssertionError as e:
+            raise InvalidExpectationConfigurationError(str(e))
+
+    # Metadata for display in the public gallery
+    library_metadata = {
+        "tags": [
+            "basic math",
+            "multi-column expectation",
+        ],
+        "contributors": ["@jorgicol", "@exteli"],
+    }
+
+if __name__ == "__main__":
+    ExpectColumnPairValuesToHaveDifferenceOfCustomPercentage().print_diagnostic_checklist()
+
+diagnostics = ExpectColumnPairValuesToHaveDifferenceOfCustomPercentage().run_diagnostics()
+
+for check in diagnostics["tests"]:
+    assert check["test_passed"] is True
+    assert check["error_diagnostics"] is None
+
+for check in diagnostics["errors"]:
+    assert check is None
+
+for check in diagnostics["maturity_checklist"]["experimental"]:
+    if check["message"] == "Passes all linting checks":
+        continue
+    assert check["passed"] is True
```

Comment:
This block can be removed; it is generally only included for integration testing on expectations referenced in certain documentation.